gobblin-metastore/src/main/java/gobblin/metastore/FsStateStore.java

```diff
@@ -174,11 +172,13 @@ public void putAll(String storeName, String tableName, Collection<T> states)
     } finally {
       closer.close();
     }
+
+    Path tablePath = new Path(new Path(this.storeRootDir, storeName), tableName);
+    HadoopUtils.renamePath(this.fs, tmpTablePath, tablePath);
```

Comment:
But it extends FsStateStore. I was wondering if @ibuenros statement about the transient nature of the data holds true for FsDatasetStateStore as well.